### PR TITLE
Temporal Nexus: adding Nexus terms in Glossary 

### PR DESCRIPTION
**Pull request by jsundai**
## What does this PR do?
This pull request brings some fresh additions to our glossary, focusing on Nexus. These are tagged with [term], [nexus], and [explanation].

Glossary Additions:
- Caller (aka source, consumer)
- Nexus Client
- Handler (aka callee, target, destination, provider)
- Nexus API Registry
- Nexus Endpoint
- Nexus Service
- Nexus Operation
- Nexus Operation Reference
- Nexus Operation Handler
- Nexus Operation Handle
- Nexus API Definitions
- Async Completion Callback
- Nexus Operation Events

Made some small spelling and grammar edits to other entries. 

## Notes to reviewers
The anchors are all placeholders right now. Not ready to merge until the anchors are in place on the pages. 
For Vale, these will need to be added.

### CODE REVIEW

File: docs/glossary.md [hunk 1]
```diff
@@ -242,6 +254,12 @@ When service to the primary Cluster is compromised, a [failover](#failover) tran
 
 _Tags: [term](/tags/term), [explanation](/tags/explanation)_
 
+#### [Handler](/nexus#handler)
```
Comment (axfelix) [inline docs/glossary.md:257]: Hmmmmm, are we generally calling this a "handler" and not a "Nexus handler"? We use "handles" a lot in Core SDKs.

File: docs/glossary.md [hunk 2]
```diff
@@ -302,10 +320,73 @@ _Tags: [term](/tags/term), [explanation](/tags/explanation)_
...
+
+#### [Nexus Client](/nexus#nexus-client)
+
+Used by the Caller to connect to a Nexus Endpoint and Nexus Service.
+
+_Tags: [term](/tags/term), [nexus](/tags/nexus), [explanation](/tags/explanation)_
+
+#### [Nexus Endpoint](/nexus#nexus-endpoint)
+
+A Nexus Endpoint is like an API proxy that can serve one or more Nexus Services from a target such as a Worker listening on a target Temporal Namespace and target Task Queue.
```
Comment (axfelix) [inline docs/glossary.md:347]: Curious (possibly stupid question) -- what makes it "like" an API proxy rather than just "an API proxy"?

**Review by MasonEgger — APPROVED** [review]
Approved, under the condition the build gets fixed.

File: docs/glossary.md (hunk 2 above)
Comment (prasek) [reply docs/glossary.md:347]: Would be fine removing "like" if we're comfortable with "API proxy" in this context translating to a  Temporal Nexus API proxy. Calling it an API proxy does make the concept immediately understandable to most everyone we have talked with.

Why "like"? Most traditional API proxies are typical HTTP request/response, whereas a Temporal Nexus Endpoint routes via Temporal's queue-based worker architecture via a namespace + task queue (and some Nexus RPC internally across some hops), so it didn't feel correct to position it as a traditional API proxy, but also it does perform the functions of an API proxy, perhaps just with a Temporal transport for part of the journey from caller to handler.

Also note that a Temporal Nexus Endpoint is more than just the low-level [Nexus endpoint](https://github.com/nexus-rpc/api/blob/30ebcf5ded186ccfc70759f864ce8a43d6c73859/SPEC.md?plain=1#L17), which does not route to Temporal namespaces, and Nexus RPC is agnostic to Temporal entirely -- much like a [Google Cloud Endpoint](https://cloud.google.com/endpoints) is a gRPC endpoint, but is also doing their own API proxy/gateway things, and gRPC is agnostic to Google Cloud Endpoints.

File: docs/glossary.md [hunk 3]
```diff
@@ -302,10 +320,73 @@ _Tags: [term](/tags/term), [explanation](/tags/explanation)_
...
+
+#### [Nexus Operation](/nexus#nexus-operation)
+
+An arbitrary-length service call that may be synchronous or asynchronous, short-lived, or long-lived, used to invoke operations across Namespace, Temporal Service, and cell boundaries and into arbitrary external systems.
+
+_Tags: [term](/tags/term), [nexus](/tags/nexus), [explanation](/tags/explanation)_
+
+#### [Nexus Operation Events](/nexus#nexus-operation-events)
+
+History events that surface in the Caller Workflow to indicate the state of an Operation including `Op Scheduled`, `Op Started`, `Op Completed`.
```
Comment (prasek) [inline docs/glossary.md:361]:
```suggestion
History events that surface in the Caller Workflow to indicate the state of an Operation including `Nexus Operation Scheduled`, `Nexus Operation Started`, `Nexus Operation Completed`.
```

File: docs/glossary.md [hunk 4]
```diff
@@ -302,10 +320,73 @@ _Tags: [term](/tags/term), [explanation](/tags/explanation)_
...
+
+_Tags: [term](/tags/term), [nexus](/tags/nexus), [explanation](/tags/explanation)_
+
+#### [Nexus Operation Handle](/nexus#nexus-operation-handle)
```
Comment (prasek) [inline docs/glossary.md:365]: Would omit this here, too low level and the examples will be SDK specific
```suggestion
```

File: docs/glossary.md [hunk 5]
```diff
@@ -302,10 +320,73 @@ _Tags: [term](/tags/term), [explanation](/tags/explanation)_
...
+
+Returned to the Workflow that started an operation to interact with the operation, for example `handle.WaitStarted()`, `handle.GetResult()`.
```
Comment (prasek) [inline docs/glossary.md:367]:
```suggestion
```

File: docs/glossary.md (hunk 1 above)
Comment (prasek) [inline docs/glossary.md:257]: would skip this one and keep the Nexus Operation Handler definition lower in the glossary.

```suggestion
```

File: docs/glossary.md [hunk 6]
```diff
@@ -242,6 +254,12 @@ When service to the primary Cluster is compromised, a [failover](#failover) tran
 
 _Tags: [term](/tags/term), [explanation](/tags/explanation)_
 
+#### [Handler](/nexus#handler)
+
+The Handler of a Nexus Operation, running in a Worker that is polling a Task Queue the target/destination Namespace, and associated underlying Temporal primitives like a Workflow. The Handler is also known as a callee, target, destination, or provider.
```
Comment (prasek) [inline docs/glossary.md:259]: as above, would skip this for the Temporal glossary

```suggestion
```

File: docs/glossary.md [hunk 7]
```diff
@@ -101,6 +107,12 @@ The `Authorizer` plugin contains a single `Authorize` method, which is invoked f
 
 _Tags: [term](/tags/term)_
 
+#### [Caller](/nexus#caller)
+
+The initiator or caller of a Nexus Operation, also known as the service consumer calling a service API. A Caller is also known as a source or consumer.
```
Comment (prasek) [inline docs/glossary.md:112]: as above, would skip this for the Temporal glossary

```suggestion
```

File: docs/glossary.md [hunk 8]
```diff
@@ -302,10 +320,73 @@ _Tags: [term](/tags/term), [explanation](/tags/explanation)_
...
+
+_Tags: [term](/tags/term), [nexus](/tags/nexus), [explanation](/tags/explanation)_
+
+#### [Nexus API Registry](/nexus#nexus-api-registry)
+
+An account-scoped list of Endpoints and Services available for use within a Temporal Cloud account, including an Endpoint description of the Services and Operations which the Endpoint supports. 
+
+_Tags: [term](/tags/term), [nexus](/tags/nexus), [explanation](/tags/explanation)_
+
+#### [Nexus Client](/nexus#nexus-client)
```
Comment (prasek) [inline docs/glossary.md:339]: Go SDK specific, would remove

```suggestion
```

File: docs/glossary.md [hunk 9]
```diff
@@ -302,10 +320,73 @@ _Tags: [term](/tags/term), [explanation](/tags/explanation)_
...
+
+A Nexus Endpoint is like an API proxy that can serve one or more Nexus Services from a target such as a Worker listening on a target Temporal Namespace and target Task Queue. 
+This allows service providers to present a clean API contract and hide the underlying implementation, which may consist of many internal Workflows. 
+Multiple Nexus Endpoints can target the same Namespace, and over time a Nexus Endpoint will be able to span multiple Namespaces with service routing rules.
+
+_Tags: [term](/tags/term), [nexus](/tags/nexus), [explanation](/tags/explanation)_
+
+#### [Nexus Operation](/nexus#nexus-operation)
+
+An arbitrary-length service call that may be synchronous or asynchronous, short-lived, or long-lived, used to invoke operations across Namespace, Temporal Service, and cell boundaries and into arbitrary external systems.
```
Comment (prasek) [inline docs/glossary.md:355]:
```suggestion
An arbitrary-duration operation that may be synchronous or asynchronous, short-lived, or long-lived, and used to connect durable executions within and across namespaces, clusters, regions, and clouds. Unlike a traditional RPC, an asynchronous Nexus Operation has an operation identity that can be used to re-attach to a long-lived Nexus Operation, for example, one backed by a Temporal Workflow. Nexus Operations support a uniform interface to get the status of an operation or its result, receive a completion callback, or cancel the operation – all of which are fully integrated into the Temporal platform.
```

File: docs/glossary.md [hunk 10]
```diff
@@ -302,10 +320,73 @@ _Tags: [term](/tags/term), [explanation](/tags/explanation)_
...
+
+_Tags: [term](/tags/term), [nexus](/tags/nexus), [explanation](/tags/explanation)_
+
+#### [Nexus Operation Handler](/nexus#nexus-operation-handler)
+
+The handler code in a Handler Worker, used to process service calls for a given Operation. Operation Handlers often run in the same Worker as the Workflows they abstract, to provide a service interface that can be used across Namespace boundaries.
+
+_Tags: [term](/tags/term), [nexus](/tags/nexus), [explanation](/tags/explanation)_
+
+#### [Nexus Operation Reference](/nexus#nexus-operation-reference)
```
Comment (prasek) [inline docs/glossary.md:377]: outdated

```suggestion
```

File: docs/glossary.md [hunk 11]
```diff
@@ -302,10 +320,73 @@ _Tags: [term](/tags/term), [explanation](/tags/explanation)_
...
+
+Key information about an Operation including the name and input/output types sufficient for invoking the Operation from a Caller Workflow, or implementing a Handler for the Operation in a Handler Worker.
```
Comment (prasek) [inline docs/glossary.md:379]: as above

```suggestion
```

File: docs/glossary.md [hunk 12]
```diff
@@ -302,10 +320,73 @@ _Tags: [term](/tags/term), [explanation](/tags/explanation)_
...
+
+_Tags: [term](/tags/term), [nexus](/tags/nexus), [explanation](/tags/explanation)_
+
+#### [Nexus Service](/nexus#nexus-service)
+
+Nexus Services are a Temporal primitive that supports arbitrary-duration Nexus Operations compatible with durable execution systems.
```
Comment (prasek) [inline docs/glossary.md:385]:
```suggestion
A Nexus Service is a named collection of arbitrary-duration Nexus Operations that provide an API contract suitable for sharing across team boundaries.
```

File: docs/glossary.md [hunk 13]
```diff
@@ -302,10 +320,73 @@ _Tags: [term](/tags/term), [explanation](/tags/explanation)_
...
+#### [Nexus Operation Reference](/nexus#nexus-operation-reference)
+
+Key information about an Operation including the name and input/output types sufficient for invoking the Operation from a Caller Workflow, or implementing a Handler for the Operation in a Handler Worker.
+
+_Tags: [term](/tags/term), [nexus](/tags/nexus), [explanation](/tags/explanation)_
+
+#### [Nexus Service](/nexus#nexus-service)
+
+Nexus Services are a Temporal primitive that supports arbitrary-duration Nexus Operations compatible with durable execution systems. 
+Unlike a traditional RPC, an arbitrary-length Nexus Operation has an identity (id) that lasts beyond the process lifetimes of the Caller and Handler, and can be used to re-attach to a long-lived Nexus Operation, for example, one backed by a Temporal Workflow.
```
Comment (prasek) [inline docs/glossary.md:386]:
```suggestion
Nexus Services are registered with a Temporal Worker that is listening on the target namespace and task queue for an Endpoint.
```

File: docs/glossary.md [hunk 14]
```diff
@@ -302,10 +320,73 @@ _Tags: [term](/tags/term), [explanation](/tags/explanation)_
...
+
+#### [Nexus Operation Handle](/nexus#nexus-operation-handle)
+
+Returned to the Workflow that started an operation to interact with the operation, for example `handle.WaitStarted()`, `handle.GetResult()`.
+
+_Tags: [term](/tags/term), [nexus](/tags/nexus), [explanation](/tags/explanation)_
+
+#### [Nexus Operation Handler](/nexus#nexus-operation-handler)
+
+The handler code in a Handler Worker, used to process service calls for a given Operation. Operation Handlers often run in the same Worker as the Workflows they abstract, to provide a service interface that can be used across Namespace boundaries.
```
Comment (prasek) [inline docs/glossary.md:373]:
```suggestion
The Nexus handler code in a Temporal Worker typically created using Temporal SDK builder functions that make it easy to abstract Temporal primitives and expose a clean service contract for others to use.
```

**Review by prasek — CHANGES_REQUESTED** [review]
dropped some suggestions, thanks!

File: docs/glossary.md (hunk 9 above)
Comment (bergundy) [reply docs/glossary.md:355]: I like what you've added here. I would change "re-attach to a long-lived Nexus Operation" to "re-attach to it".

File: docs/glossary.md (hunk 3 above)
Comment (bergundy) [reply docs/glossary.md:361]: We don't have a section for activity events, I don't think we need one for Nexus.

File: docs/glossary.md (hunk 10 above)
Comment (bergundy) [reply docs/glossary.md:377]: It's not outdated, it's just a Go specific concept.

File: docs/glossary.md [hunk 15]
```diff
@@ -101,6 +107,12 @@ The `Authorizer` plugin contains a single `Authorize` method, which is invoked f
 
 _Tags: [term](/tags/term)_
 
+#### [Caller](/nexus#caller)
```
[the remark this answers is not in the document]
Comment (jsundai) [reply docs/glossary.md:110]: removing

File: docs/glossary.md (hunk 1 above)
Comment (jsundai) [reply docs/glossary.md:257]: removing

File: docs/glossary.md (hunk 8 above)
Comment (jsundai) [reply docs/glossary.md:339]: removing

File: docs/glossary.md (hunk 3 above)
Comment (jsundai) [reply docs/glossary.md:361]: It doesn't hurt to have more definitions & it helps out with Ask AI purposes.

**Comment by jsundai** [comment]
> Approved, under the condition the build gets fixed.

Yeah it might be simplest to publish this first and then add another PR with the anchors/links once everything is published.

File: docs/glossary.md [hunk 16]
```diff
@@ -302,10 +308,60 @@ _Tags: [term](/tags/term), [explanation](/tags/explanation)_
...
+
+_Tags: [term](/tags/term), [nexus](/tags/nexus), [explanation](/tags/explanation)_
+
+#### [Nexus Service](/nexus#nexus-service)
+
+A Nexus Service is a named collection of arbitrary-duration Nexus Operations that provide an API contract suitable for sharing across team boundaries.
+Nexus Services are registered with a Temporal Worker that is listening on the target Namespace and Task Queue for an Endpoint.
+
+_Tags: [term](/tags/term), [nexus](/tags/nexus), [explanation](/tags/explanation)_
+
```
Comment (prasek) [inline docs/glossary.md:364]:
```suggestion
#### [Nexus RPC](https://github.com/nexus-rpc/api/blob/main/SPEC.md)
Nexus RPC is a new protocol designed with durable execution in mind. It supports arbitrary-duration Operations that extend beyond a traditional RPC — a key underpinning to connect durable executions within and across namespaces, clusters, regions, and clouds.
```

Comment (prasek) [inline docs/glossary.md:364]: need to check the link, but that page has details on the Nexus Machinery

```suggestion
#### [Nexus Machinery](/cloud/nexus/using-nexus)
In Temporal Cloud, the Nexus Machinery routes Nexus requests across Namespace boundaries via a secure mTLS Envoy mesh, where they are sync matched to a Worker in the handler Namespace, which processes the Nexus request.
```

File: docs/glossary.md (hunk 2 above)
Comment (jsundai) [reply docs/glossary.md:331]: Thanks for the context. Going to removing the word "like" for now unless there's an alternative.

**Pull request merged**